photon_installer/installer.py

```diff
@@ -926,10 +926,28 @@ def _setup_install_repo(self):
     def _install_additional_rpms(self):
         rpms_path = self.install_config.get('additional_rpms_path', None)
 
-        if not rpms_path or not os.path.exists(rpms_path):
+        rpm_dir = []
+        if not rpms_path or len(rpm_dir) == 0:
             return
 
-        if self.cmd.run(['rpm', '--root', self.photon_root, '-U', rpms_path + '/*.rpm']) != 0:
+        if isinstance(rpms_path, list):
+            rpm_dir = [d for d in rpm_dir if len(d) > 0
+                       and os.path.exists(d)]
```

Comment:
So I specifically did this because there is an issue with how cmd returns the exit code. The exit code that we use now needs to be a sufficient condition, and I did a separate patch for that.  In essence, rpm can execute, but it returns an exit code that indicates something different than an error.   i.e., it is not a reliable indication.   ( I need to cross-check my notes.  but consider the same RPM, i.e., something already installed,   something requires another RPM.
(From my memory, I'll check the rpm return for all these case exit codes!= 0 )

Please also please check this because it is related if an exception is raised.  We have no exit code that leads to chain issues.  

```
 process = subprocess.Popen(
                cmd, shell=use_shell,
                stdout=subprocess.PIPE,
                stderr=subprocess.PIPE
            )
```
can return FileNotFoundError
https://github.com/vmware/photon-os-installer/pull/18/commits/3431f6fba3a3521d3d562ee96348b64ed2a23a27


That way, I consider this as the best effort.  "Attempts" since it is additional rpms and technically, you can finish the installation without crashing the installer.